src/server/req_delete.c

```diff
@@ -1627,19 +1627,27 @@ post_delete_mom1(struct work_task *pwt)
 				return;
 			}
 			/* 2nd try, use SIGTERM */
-			rc = issue_signal(pjob, sigt, post_delete_mom1, preq_clt);
-			if (rc == 0)
-				return; /* will be back when replies */
-			goto resend;
+			/* prevent infinite loop, try 10 times only */
+			if (preq_clt->rq_reply.brp_count < 10) {
+				preq_clt->rq_reply.brp_count++;
+				rc = delayed_issue_signal(pjob, sigt, post_delete_mom1, preq_clt, 5);
```

Comment:
Consider this delay of 5 seconds be configurable as a server attribute. 5 seconds may not always be the best. It might need more.

Reply:
The new server attribute `resend_term_delay` was added.